pkg/objstore/inmem/inmem.go

```diff
@@ -16,7 +17,9 @@ import (
 var errNotFound = errors.New("inmem: object not found")
 
 // Bucket implements the store.Bucket and shipper.Bucket interfaces against local memory.
+// methods from Bucket interface are thread-save. Object are assumed immutable.
 type Bucket struct {
+	mtx     sync.RWMutex
```

Comment:
Nothing is ever write-locking this mutex, so what is this mutex actually protecting against?